core/src/main/java/com/digitalpebble/stormcrawler/filtering/sitemap/SitemapFilter.java

```diff
@@ -40,15 +42,17 @@
 public class SitemapFilter implements URLFilter {
 
     @Override
-    public String filter(URL sourceUrl, Metadata sourceMetadata, String urlToFilter) {
-        if (sourceMetadata != null) {
-            if ("false"
-                            .equalsIgnoreCase(
-                                    sourceMetadata.getFirstValue(SiteMapParserBolt.isSitemapKey))
-                    && "true"
-                            .equalsIgnoreCase(
-                                    sourceMetadata.getFirstValue(
-                                            SiteMapParserBolt.foundSitemapKey))) return null;
+    public @Nullable String filter(
+            @Nullable URL sourceUrl,
+            @Nullable Metadata sourceMetadata,
+            @NotNull String urlToFilter) {
+
+        if (sourceMetadata != null
+                && !Boolean.parseBoolean(
```

Comment:
Should't we be consistent in the project? Other sources use the explicit form

Reply:
What do you mean with explicit form?
As far as i know that is the "standard java way" or at least that what you should use in this situation.

Reply:
In the other situation from before, we have a distinction between "true", "false" and null/invalid. Therefore we had to distinct them. Here we have that it is either "true" or whatever.

Reply:
Sometimes the PR uses `true`.equalsIgnoreCase(...) and sometimes `Boolean.parseBoolean(...)`  ;-) - so i was thinking about consistency

Reply:
> In the other situation from before, we have a distinction between "true", "false" and null/invalid. Therefore we had to distinct them. Here we have that it is either "true" or whatever.

Yup - I can live with both versions. It was just about consistency :)

Reply:
Yes, but at some places I left out many useful/correct changes by intent.
Because this PR is already so big that it is practically a crime.

If you take a look at my fork (https://github.com/FelixEngl/storm-crawler/tree/local_version) you can see what I plan to add PR by PR. (except the changes of the pom.xml for installing it to my local repo.)

If you want/have the muse for it you can also give me feedback there if you see something you don't like or want changed. :)

Reply:
> If you take a look at my fork (https://github.com/FelixEngl/storm-crawler/tree/local_version) you can see what I plan to add PR by PR. (except the changes of the pom.xml for installing it to my local repo.)
> 
> If you want/have the muse for it you can also give me feedback there if you see something you don't like or want changed. :)

I have run SC for 370 days 24/7 with only a few downtimes for gathering PhD related data. The code per-se (and storm) is pretty stable but I am interested in the development of SC for further research questions. So yes - I am happy to have a look but we would need to enable some sort of communication on your fork. If you like, you can also drop a mail at my university mail address (should be obvious to find), so we can get in touch ;)